src/core/components/dag.js

```diff
@@ -73,6 +75,31 @@ module.exports = function dag (self) {
         self._ipld.treeStream(cid, path, options),
         pull.collect(callback)
       )
+    }),
+
+    // TODO - move to IPLD resolver and generalize to other IPLD formats
+    _getRecursive: promisify((multihash, callback) => {
+      // gets flat array of all DAGNodes in tree given by multihash
+      callback = once(callback)
+      self.dag.get(new CID(multihash), (err, res) => {
+        if (err) { return callback(err) }
+        const links = res.value.links
+        const nodes = [res.value]
+        // leaf case
+        if (!links.length) {
+          return callback(null, nodes)
+        }
+        // branch case
+        links.forEach(link => {
+          self.dag._getRecursive(link.multihash, (err, subNodes) => {
+            if (err) { return callback(err) }
+            nodes.push(subNodes)
+            if (nodes.length === links.length + 1) {
+              return callback(null, _.flattenDeep(nodes))
+            }
+          })
+        })
+      })
```

Comment:
@vmx can we get this API call on http://github.com/ipld/js-ipld?

Reply:
@vmx Will selectors enable something like `<roothash>/**/*`?

Reply:
@JonKrone Selectors don't really exist yet.